proxy/upgrade.go

```diff
@@ -124,7 +124,12 @@ func (p *upgradeProxy) serveHTTP(w http.ResponseWriter, req *http.Request) {
 	if resp.StatusCode != http.StatusSwitchingProtocols {
 		log.Debugf("Got invalid status code from backend: %d", resp.StatusCode)
 		w.WriteHeader(resp.StatusCode)
-		w.Write([]byte(http.StatusText(resp.StatusCode)))
+		_, err := io.Copy(w, resp.Body)
+		if err != nil {
+			log.Errorf("Error writing body to client: %s", err)
+			return
+		}
+		resp.Body.Close()
```

Comment:
we should call resp.Body.Close regardless if the copy succeeded or failed.

Nice to have: we could close the response body in each case where we return from here.

Reply:
Fixed! Thanks! :)